src/client/opamClient.ml

```diff
@@ -2688,7 +2690,8 @@ module PIN = struct
           OpamStd.Sys.exit_because `Aborted
       | None ->
         OpamConsole.error_and_exit `Not_found
-          "Package is not pinned, and no existing version was supplied."
+          "Package is not pinned, and no existing version was supplied.%s" 
+          (OpamSwitchState.did_you_mean st [name, None])
```

Comment:
Actually, the hint here is given based on _all known_ packages and not only _st.pinned_ ones which could be confusing. I could generalise the `installed_only` argument of `did_you_mean` into `kind` (`All_known | Installed | Pinned `), would that be desirable?